prefect_shell/commands.py

```diff
@@ -110,3 +118,284 @@ def example_shell_run_command_flow():
+        """
+        Triggers a shell command and returns the shell command run object
+        to track the execution of the run. This method is ideal for long-lasting
+        shell commands; for short-lasting shell commands, it is recommended
+        to use the `run` method instead.
+
+        Args:
+            **open_kwargs: Additional keyword arguments to pass to `open_process`.
+
+        Returns:
+            A `ShellProcess` object.
+
+        Examples:
+            Sleep for 5 seconds and then print "Hello, world!":
+            ```python
+            from prefect_shell import ShellOperation
+
+            with ShellOperation(
+                commands=["sleep 5", "echo 'Hello, world!'"],
+            ) as shell_operation:
+                shell_process = await shell_operation.trigger()
+                shell_process.wait_for_completion()
+                shell_output = shell_process.fetch_result()
+            ```
+        """
+        extension = self.extension or (".ps1" if sys.platform == "win32" else ".sh")
+        temp_file = self._exit_stack.enter_context(
+            tempfile.NamedTemporaryFile(
+                prefix="prefect-",
+                suffix=extension,
+            )
+        )
+
+        joined_commands = os.linesep.join(self.commands)
+        self.logger.debug(
+            f"Writing the following commands to "
+            f"{temp_file.name!r}:{os.linesep}{joined_commands}"
+        )
+        temp_file.write(joined_commands.encode())
+        temp_file.flush()
+
+        if self.shell is None and sys.platform == "win32" or extension == ".ps1":
+            shell = "powershell"
+        elif self.shell is None:
+            shell = "bash"
+        else:
+            shell = self.shell.lower()
+
+        if shell == "powershell":
+            # if powershell, set exit code to that of command
+            temp_file.write("\r\nExit $LastExitCode".encode())
+
+        num_commands = len(self.commands)
+        trigger_command = [shell, temp_file.name]
+        input_env = os.environ.copy()
+        input_env.update(self.env)
+        self.logger.debug(f"Preparing to execue {trigger_command}")
+        process = await self._exit_stack.enter_async_context(
+            open_process(
+                trigger_command,
+                stdout=subprocess.PIPE if self.stream_output else subprocess.DEVNULL,
+                stderr=subprocess.PIPE if self.stream_output else subprocess.DEVNULL,
+                env=input_env,
+                cwd=self.working_dir,
+                **open_kwargs,
+            )
+        )
+        self.logger.info(
+            f"PID {process.pid} triggered with {num_commands} commands running "
+            f"inside the {(self.working_dir or '.')!r} directory."
+        )
+        return ShellProcess(shell_operation=self, process=process)
+
+    @sync_compatible
+    async def run(self, **open_kwargs: Dict[str, Any]) -> List[str]:
```

Comment:
For this method, can we do the context management ourselves? It seems like a bit much to have a context manager for a short-running command or script.